src/test/java/co/rsk/federate/signing/hsm/client/HSMClientVersion2BTCTest.java

```diff
@@ -445,6 +437,34 @@ public void updateAncestorBlock_ok() throws HSMClientException, JsonRpcException
         Assert.assertEquals("updateAncestorBlock", capturedArguments.get(2).get("command").asText());
     }
 
+    @Test
+    public void updateAncestorBlock_hsm_version_3() throws HSMClientException, JsonRpcException {
+        JsonRpcClientProvider jsonRpcClientProvider = mock(JsonRpcClientProvider.class);
+        JsonRpcClient jsonRpcClient = mock(JsonRpcClient.class);
+        when(jsonRpcClientProvider.acquire()).thenReturn(jsonRpcClient);
+
+        HSMClientProtocol hsmClientProtocol = new HSMClientProtocol(jsonRpcClientProvider, ECDSASignerFactory.DEFAULT_ATTEMPTS, ECDSASignerFactory.DEFAULT_INTERVAL);
+        HSMClientVersion2BTC client = new HSMClientVersion2BTC(hsmClientProtocol, 3);
+        client.setMaxChunkSizeToHsm(2);
+
+        when(jsonRpcClient.send(any(JsonNode.class))).thenReturn(buildResponse(0));
+        when(jsonRpcClient.send(hsmClientProtocol.buildCommand("blockchainState", 3)))
+            .thenReturn(buildResponse(0, "state", buildStateResponse(false)));
+
+        BlockHeader blockHeader = mock(BlockHeader.class);
+        when(blockHeader.getEncoded(true, false)).thenReturn(new byte[]{});
+
+        client.updateAncestorBlock(new UpdateAncestorBlockMessage(Arrays.asList(blockHeader, blockHeader, blockHeader)));
+        ArgumentCaptor<JsonNode> captor = ArgumentCaptor.forClass(JsonNode.class);
+        verify(jsonRpcClient, times(3)).send(captor.capture());
+        List<JsonNode> capturedArguments = captor.getAllValues();
+        Assert.assertEquals("blockchainState", capturedArguments.get(0).get("command").asText());
+        Assert.assertEquals("updateAncestorBlock", capturedArguments.get(1).get("command").asText());
```

Comment:
Why is updateAncestorBlock called twice?

Reply:
The BlockHeaders Was Broken Into 2 Chunks.. Hence, Calling The updateAncestorBlock Twice

Reply:
I see, thanks. Might be worth adding a comment then. Or changing the test name

Reply:
Okay, Done.